pxtblocks/blocklyloader.ts

```diff
@@ -913,21 +1058,168 @@ namespace pxt.blocks {
+        installHelpResources(
+            'controls_if',
+            lf("a conditional statement"),
+            undefined,
+            "blocks/logic/if"
+        );
+
+        // builtin logic_compare
+        Blockly.Blocks['logic_compare'].init = function() {
+            let OPERATORS = [
+                [lf("=="),  'EQ'],  // =
+                [lf("_op_neq"), 'NEQ'], // ≠
+                [lf("_op_lt"),  'LT'],  // <  (ltr)
+                [lf("_op_lte"), 'LTE'], // ≦ (ltr)
+                [lf("_op_gt"),  'GT'],  // >  (ltr)
+                [lf("_op_gte"), 'GTE']  // ≧ (ltr)
+            ];
+            this.setColour(blockColors['logic']);
+            this.setOutput(true, 'Boolean');
+            this.appendValueInput('A');
+            this.appendValueInput('B')
+                .appendField(new Blockly.FieldDropdown(OPERATORS), 'OP');
+            this.setInputsInline(true);
+            this.prevBlocks_ = [null, null];
+
+            let TOOLTIPS:any = {
+                'EQ': lf("Return true if both inputs equal each other."),
+                'NEQ': lf("Return true if both inputs are not equal to each other."),
+                'LT': lf("Return true if the first input is smaller than the second input."),
+                'LTE': lf("Return true if the first input is smaller than or equal to the second input."),
+                'GT': lf("Return true if the first input is greater than the second input."),
+                'GTE': lf("Return true if the first input is greater than or equal to the second input.")
+            };
+
+            var thisBlock = this;
+            setHelpResources(this,
+                'logic_compare',
+                lf("comparing two numbers"),
+                function() {
+                    return TOOLTIPS[thisBlock.getFieldValue('OP')];
+                },
+                '/blocks/logic/boolean'
+            );
+        };
+
+        // builtin logic_operation
+        msg.LOGIC_OPERATION_AND = lf("&&");
+        msg.LOGIC_OPERATION_OR = lf("||");
+        msg.LOGIC_OPERATION_TOOLTIP_AND = lf("Return true if both inputs are true."),
+        msg.LOGIC_OPERATION_TOOLTIP_OR = lf("Return true if at least one of the inputs is true."),
+        installHelpResources(
+            'logic_operation',
+            lf("boolean operation"),
+            undefined,
+            '/blocks/logic/boolean'
+        );
+
+        // builtin logic_negate
+        msg.LOGIC_NEGATE_TITLE = lf("not %1");
+        installHelpResources(
+            'logic_negate',
+            lf("logical negation"),
+            lf("Returns true if the input is false. Returns false if the input is true."),
+            '/blocks/logic/boolean'
+        );
+
+        // builtin logic_boolean
+        msg.LOGIC_BOOLEAN_TRUE = lf("_logic_true");
```

Comment:
Same as ``_for_do`` and ``_while_do``, please open an issue and we'll fix this in a future commit.